### DISCUSSION TITLE
Dotnet xxx.Extensions.Caching.Nats

### DISCUSSION THREAD

**Discussion by alberk8** [nats-io/nats.net#431] [Ideas]
I would really like if there is an official nuget to support caching as an Extension  much like `Microsoft.Extensions.Caching.StackExchangeRedis`

**Comment by mtmk** [comment]
I think this is already being looked into by @stebet

**Reply by alberk8** [reply]
I want to be first to test it out :).

I want to be able to consolidate RabbitMQ and Redis to NATS asap.

**Reply by alberk8** [reply]
Oh, if someone is adding the Cache for SignalR it would be wonderful.

**Reply by mtmk** [reply]
why not you? 😝 😄

**Reply by alberk8** [reply]
I am not good at writing class library.  That is the issue 😂.

**Reply by stebet** [reply]
Waiting for the first builds with TTL per key to arrive and then I'll go all in on doing this. 

I've already implemented a NATS backplane for SignalR at work, need to clean that up and put in a package :)

**Reply by alberk8** [reply]
When is the TTL per key coming to NATS.  Will the keys still be in a bucket like what it is now?.    I like the SignalR backplane with NATS.   Then I can have just NATS for all the Cache and messaging.

**Reply by mtmk** [reply]
ttl per key is planned for nats-server 2.11 keep an [eye on the releases](https://github.com/nats-io/nats-server/releases)

**Reply by alberk8** [reply]
I have being checking every single day since. :)

**Reply by yeongjonglim** [reply]
Hi @mtmk, as [2.11](https://github.com/nats-io/nats-server/releases/tag/v2.11.1-binary) has been released, is anyone still looking to implement the interface `IDistributedCache` on NATS KV store? I would like to help if no one is already working on it.

**Reply by alberk8** [reply]
There is one [here ](https://github.com/stebet/Stebet.SignalR.NATS)
For [SignalR](https://github.com/stebet/Stebet.SignalR.NATS)  
Both the nugets are still behind.

**Reply by mtmk** [reply]
@stebet and @jodydonetti are looking into it. we are waiting for a patch release to the server for official support for KV TTL stuff fully. so the server is not 100% there yet.

**Reply by stebet** [reply]
It's coming :) Like @mtmk said we are waiting for a patch release for 2.11 to release it properly. You can test the prerelease of the IDistributedCache implementation already to prepare.

Then I'm also working with @jodydonetti for NATS support in FusionCache, both as an IDistributedCache as well as a backplane. Stay tuned for that :)

**Reply by jodydonetti** [reply]
Hi all, all I can say is... damn, you can't imagine all the goodness that's coming 😬

**Reply by alberk8** [reply]
is there any update here?  Thanks.
